rubin_sim/scheduler/basis_functions/basis_functions.py

```diff
@@ -1907,30 +1907,46 @@ class VisitGap(Base_basis_function):
     ----------
     note : str
         Value of the observation "note" field to be masked.
+    filter_names : list [str], optional
+         List of filter names that need be observed before activating.
```

Comment:
"Before activating" -- would it also make sense to reword and extend this as 
'List of filter names that will be considered when evaluating if the `gap` has passed. All filters must be observed before the gap requirement will be activated, and once activated, only observations in these filters will be evaluated in context of whether the last observation was at least `gap` in the past.'

Reply:
I kept your first phrase in the parameter docstring and moved everything else to a Notes section.